src/puppetlabs/puppetdb/command.clj

```diff
@@ -214,12 +215,17 @@
 ;; Node deactivation
 
 (defmethod process-command! [(command-names :deactivate-node) 2]
-  [{certname :payload {:keys [id]} :annotations} {:keys [db]}]
-  (jdbc/with-transacted-connection db
-    (when-not (scf-storage/certname-exists? certname)
-      (scf-storage/add-certname! certname))
-    (scf-storage/deactivate-node! certname))
-  (log/info (format "[%s] [%s] %s" id (command-names :deactivate-node) certname)))
+  [{:keys [payload annotations]} {:keys [db]}]
+  (let [certname (:certname payload)
+        producer_timestamp (to-timestamp (or (:producer_timestamp payload) (now)))
```

Comment:
as in `(get payload :producer_timestamp (now))`? I always forget that's a
thing you can do, will fix.

On Wed, Apr 22, 2015 at 4:58 PM, Ryan Senior <notifications@github.com>
wrote:

> In src/puppetlabs/puppetdb/command.clj
> <https://github.com/puppetlabs/puppetdb/pull/1361#discussion_r28926288>:
>
> > @@ -214,12 +215,17 @@
> >  ;; Node deactivation
> >
> >  (defmethod process-command! [(command-names :deactivate-node) 2]
> > -  [{certname :payload {:keys [id]} :annotations} {:keys [db]}]
> > -  (jdbc/with-transacted-connection db
> > -    (when-not (scf-storage/certname-exists? certname)
> > -      (scf-storage/add-certname! certname))
> > -    (scf-storage/deactivate-node! certname))
> > -  (log/info (format "[%s] [%s] %s" id (command-names :deactivate-node) certname)))
> > +  [{:keys [payload annotations]} {:keys [db]}]
> > +  (let [certname (:certname payload)
> > +        producer_timestamp (to-timestamp (or (:producer_timestamp payload) (now)))
>
> why not use the (now) as a default value and avoid the or?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetdb/pull/1361/files#r28926288>.
>